Version improvements

Holzhaus opened the pull request:
- Make version number accessible as QVersionNumber
- Show `git-describe` version in about dialog
- remove git commit count from CMakeLists
- Use git describe for package file name
- Fix quoting when logging build details

Holzhaus committed:
DlgAbout: Show git-describe version

Holzhaus committed:
CMake: Use git-describe for package file name

Holzhaus committed:
VersionStore: Fix quoting when logging build details

Holzhaus commented on src/util/versionstore.cpp:
```diff
@@ -55,21 +71,17 @@ QString VersionStore::applicationName() {
 QString VersionStore::applicationTitle() {
 #ifdef __APPLE__
```
Do we really need the distinction here? I think the application title should always be just `Mixxx`.

daschuer replied:
Yes, I agree. We just need to make sure that this info is inside the about box.

Holzhaus replied:
I removed `applicationTitle` and use `applicationName` instead. The platform information has been improved and is shown in the About dialog now.

Holzhaus commented on CMakeLists.txt:
```diff
@@ -2719,9 +2698,9 @@ string(REPLACE "\n" "\n " CPACK_DEBIAN_PACKAGE_DESCRIPTION_MERGED "${CPACK_DEBIA
 string(REPLACE "-" "~" CPACK_DEBIAN_UPSTREAM_VERSION "${MIXXX_VERSION}")
 set(CPACK_DEBIAN_DEBIAN_VERSION 0ubuntu1)
-set(CPACK_DEBIAN_DEBIAN_VERSION_EXTRA "${GIT_BRANCH}~git${GIT_COMMIT_COUNT}")
+set(CPACK_DEBIAN_DEBIAN_VERSION_EXTRA "${GIT_BRANCH}~git${GIT_DESCRIBE}")
```
Are there special requirements that might prevent usage of git describe? If so, we should document them as a comment.

daschuer replied:
Yes there are. New versions numbers have to be "greater" and there are no dashes allowed. 
I have original planned to dive into it and come up with a proposal.

daschuer replied:
How about decouple this question from this PR and keep GIT_COMMIT_COUNT unit we have everything sorted out.

daschuer replied:
The version test is: 
```
dpkg --compare-versions 2.0.0~rc1-0ubuntu1 lt 2.0.0-0ubuntu1 && echo true
```

daschuer replied:
> How about decouple this question from this PR and keep GIT_COMMIT_COUNT unit we have everything sorted out.

Or I can propose a PR against this branch. What do you think?

Holzhaus replied:
I just removed the commit from this PR and we can take care of it in a separate PR.

Holzhaus reviewed (commented):
Out of scope for this PR, but will come as a follow up: Use the new QVersionNumber for manual URLs, etc (superseding #3792).

daschuer commented on CMakeLists.txt:
```diff
@@ -2700,7 +2679,7 @@ list(APPEND CPACK_SOURCE_IGNORE_FILES "/build/")
 list(APPEND CPACK_SOURCE_IGNORE_FILES "/\\\\.sconf_temp/")
 list(APPEND CPACK_SOURCE_IGNORE_FILES "${CMAKE_CURRENT_BINARY_DIR}/")
-set(CPACK_SOURCE_PACKAGE_FILE_NAME "${CPACK_PACKAGE_FILE_NAME}-Source")
+set(CPACK_SOURCE_PACKAGE_FILE_NAME "${CPACK_PACKAGE_FILE_NAME}-src")
```
what is the reason for this change?

Holzhaus replied:
Let's use lowercase file names. This makes it less confusing when dealing with case-(in)sensitive file systems. I changed it to `source` again (I don't actually care if it's `src` or `source`).

daschuer commented:
Git describe seems to be failing on our CI:


Error: /home/runner/work/mixxx/mixxx/src/util/versionstore.cpp:41:30: error: Use an empty QLatin1String instead of an empty QStringLiteral [-Wclazy-empty-qstringliteral]
const QString kGitDescribe = QStringLiteral(GIT_DESCRIBE);
                             ^
/usr/include/x86_64-linux-gnu/qt5/QtCore/qstringliteral.h:68:9: note: expanded from macro 'QStringLiteral'
        static const QStaticStringData<Size> qstring_literal = { \
        ^
1 error generated.

Swiftb0y commented:
> Git describe seems to be failing on our CI:

See: https://github.com/mixxxdj/mixxx/pull/3259/files#diff-680cc13af33346625cdd5b4d0f392ae801b86a3a7b3c60a9f1bb729d4a28e5e1R17

daschuer commented:
I have just played a bit with it and it works nice. Thank you. 
During a release, we have a release candidate, where the CMake version has been advanced but we have no tag yet that is acceptable IMHO. I don't think we need to introduce some magic in this case.

daschuer commented:
How do we trigger the deployment of a release? 
We need to tag the release commit and build it again. Than we need to start the deployment batches. 
On Jenkins that was a build job that was started manually. Does GitHub Actions provide such manual stated jobs?

Holzhaus committed:
VersionStore: Replace applicationTitle() method with platform()

Always show the application name in the window title (without platform
information). The platform information is now shown in the About dialog.

Holzhaus committed:
CI: Add comment regarding the git fetch depth

Holzhaus commented:
> How do we trigger the deployment of a release?
> We need to tag the release commit and build it again. Than we need to start the deployment batches.
> On Jenkins that was a build job that was started manually. Does GitHub Actions provide such manual stated jobs?

I'm not sure what you mean, but it seems unrelated to this PR so let's rather discuss this on Zulip.

daschuer commented on CMakeLists.txt:
```diff
@@ -2685,7 +2687,11 @@ set(CPACK_PACKAGE_INSTALL_DIRECTORY "Mixxx")
+if(NOT GIT_DESCRIBE)
+  set(CPACK_PACKAGE_FILE_NAME "mixxx-${CMAKE_PROJECT_VERSION}-unknown")
+else()
+  set(CPACK_PACKAGE_FILE_NAME "mixxx-${GIT_DESCRIBE}")
```
Do we need to change anything our website because of this?

Holzhaus replied:
No. We don't use the regular package filename anyway (they are renamed by the `deploy.py` script):
https://github.com/mixxxdj/mixxx/blob/main/.github/workflows/build.yml#L329-L331

daschuer reviewed (approved):
LGTM, Thank you very much.
